### PR TITLE
HomeWork5

### PR DESCRIPTION
**Pull request by thejohnsondev**
In the FragmentMovieList added RecyclerView, created an adapter for it. The recycleViewr added three more films, the data for which is taken from the created Movie.class. Clicks listener on movie cards has been implemented.

### CODE REVIEW

File: app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt [hunk 1]
```diff
@@ -1,35 +1,73 @@
...
                         R.anim.slide_out
                     )
-                    replace(R.id.main_container, FragmentMoviesDetails())
                     addToBackStack(null)
+                    replace(R.id.main_container, FragmentMoviesDetails())
                     commit()
                 }
+                else -> Toast.makeText(
+                    view?.context,
+                    "Information about ${movie.name} missed",
```
Comment (emartynov) [inline app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt:54]: To string resources?

File: app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt [hunk 2]
```diff
@@ -1,35 +1,73 @@
...
                         R.anim.fade_in,
                         R.anim.slide_out
                     )
-                    replace(R.id.main_container, FragmentMoviesDetails())
                     addToBackStack(null)
+                    replace(R.id.main_container, FragmentMoviesDetails())
                     commit()
                 }
+                else -> Toast.makeText(
+                    view?.context,
```
Comment (emartynov) [inline app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt:53]: This probably will crash if view is null

File: app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt [hunk 3]
```diff
@@ -1,35 +1,73 @@
...
 import android.widget.Button
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.snackbar.Snackbar
 
 class FragmentMoviesList : Fragment() {
+
+    private var spanCount: Int? = null
```
Comment (emartynov) [inline app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt:17]: Why save it?

File: app/src/main/java/com/johnsondev/doboshacademyapp/Movie.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+data class Movie(
+    val id: Int,
+    val name: String,
+    val genre: String,
+    val reviews: Int,
+    val rating: Int,
+    val time: Int,
+    val age: Int,
+    val movieImg: Int,
```
Comment (emartynov) [inline app/src/main/java/com/johnsondev/doboshacademyapp/Movie.kt:11]: @DrawableRes

File: app/src/main/java/com/johnsondev/doboshacademyapp/MoviesAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+) : RecyclerView.Adapter<MovieViewHolder>() {
+
+    private val inflater: LayoutInflater = LayoutInflater.from(context)
+
+    override fun getItemCount(): Int = Movie.movies.size
+
+    fun getItem(position: Int): Movie = Movie.movies[position]
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {
+        val itemView = inflater.inflate(R.layout.movie_rv_item, parent, false)
```
Comment (emartynov) [inline app/src/main/java/com/johnsondev/doboshacademyapp/MoviesAdapter.kt:26]: It is safer to get context from parent

File: app/src/main/java/com/johnsondev/doboshacademyapp/MoviesAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    private val rating: RatingBar = view.findViewById(R.id.movie_rating)
+    private val time: TextView = view.findViewById(R.id.movie_time)
+    private val age: TextView = view.findViewById(R.id.movie_age)
+    private val movieImg: ImageView = view.findViewById(R.id.movie_img)
+    private val isFavorite: ImageView = view.findViewById(R.id.is_favorite_red_img)
+
+    fun bind(movie: Movie) {
+        name.text = movie.name
+        genre.text = movie.genre
+        reviews.text = "${movie.reviews} REVIEWS"
```
Comment (emartynov) [inline app/src/main/java/com/johnsondev/doboshacademyapp/MoviesAdapter.kt:53]: To string resources please

**Review by emartynov — COMMENTED** [review]
What is movie rv means? RV - RecyclerView?

A couple of comments and rest is LGTM!

File: app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt (hunk 1 above)
Comment (thejohnsondev) [reply app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt:54]: Ответ на вопрос выше: RV это типа RecyclerView, также и TV - TextView. Многие так делают, и это удобно.

File: app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt (hunk 2 above)
Comment (thejohnsondev) [reply app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt:53]: Возможно, поправлю

File: app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt (hunk 3 above)
Comment (thejohnsondev) [reply app/src/main/java/com/johnsondev/doboshacademyapp/FragmentMoviesList.kt:17]: Действительно, можно и не сейвить)

File: app/src/main/java/com/johnsondev/doboshacademyapp/Movie.kt (hunk 4 above)
Comment (thejohnsondev) [reply app/src/main/java/com/johnsondev/doboshacademyapp/Movie.kt:11]: Не знал что так нужно)

File: app/src/main/java/com/johnsondev/doboshacademyapp/MoviesAdapter.kt (hunk 6 above)
Comment (thejohnsondev) [reply app/src/main/java/com/johnsondev/doboshacademyapp/MoviesAdapter.kt:53]: Создать строковый ресурс в strings.xml для REVIEWS?

Comment (emartynov) [reply app/src/main/java/com/johnsondev/doboshacademyapp/MoviesAdapter.kt:53]: "%s Reviews"

**Comment by emartynov** [comment]
Any updates here?

**Comment by thejohnsondev** [comment]
Yes

**Pull request merged**